mypy/test/testutil.py

```diff
@@ -0,0 +1,83 @@
+from typing import Iterator, List, Tuple
+import time
+import os
+import sys
+import tempfile
+from contextlib import contextmanager
+from threading import Thread
+from unittest import TestCase, main, skipUnless
+from mypy import util
+from mypy.build import random_string
+
+
+WIN32 = sys.platform.startswith("win")
+
+
+def lock_file(filename: str, duration: float) -> Thread:
+    '''
+    Opens filename (which must exist) for reading
+    After duration sec, releases the handle
+    '''
+    def _lock_file() -> None:
+        with open(filename):
+            time.sleep(duration)
+    t = Thread(target=_lock_file, daemon=True)
+    t.start()
+    return t
+
+
+@skipUnless(WIN32, "only relevant for Windows")
+class ReliableReplace(TestCase):
+    # will be cleaned up automatically when this class goes out of scope
```

Comment:
Actually that would depend on Python finalization order which is a nightmare.

Can't you make this more explicit? E.g. in tearDownClass() below.